Summarized '#sec:rse-specialisations' (#375)
the-teachingRSE-project/competencies: summarised_competencies.md

an overview role, caring about the overall perspective (\gls{PM})
and delegating tasks to project members (e.g. \gls{USERS}, and \gls{TEAM}).
This role may or may not involve programming.

#### Teaching RSEs
Teaching (\gls{TEACH}) the next generation of scientists (RSEs or not) the core RSE skills while respecting the domain's culture requires being versed in didactics and pedagogy. Educators play
a vital role in improving the quality of research software.

#### User interface/user experience designers for research software
Good software is not only functional, but is also documented (\gls{DOCBB}) and distributed (\gls{DIST}, \ac{FAIR}). It is also designed with users in mind (\gls{MOD}).

## Specialisations outside the core RSE competencies

<!--
Research focused specialisations
-->

#### \${DOMAIN}-RSE
Some RSEs may be particularly specialized into the intricacies of a research domain,
(e.g. medical, humanities, physics).

#### Data-focused RSE
Certain domains have a high demand for data science skills.
This may be cleaning, sorting, analyzing and documenting data (\ac{RDM}, \ac{DMP}),
perhaps anonymising or pseudonymising them (e.g. patient information in medicine).

> and most often transferring securely and archiving properly \ac{FAIR} [@FAIR].

Author: ```suggestion
This role also includes transferring data sets securely and archiving them properly \ac{FAIR} [@FAIR].
```

Since \ac{ML} techniques are inherently prone to biases leading to socially problematic consequences, this calls for additional caution, especially in training data.

#### Legacy RSEs
While programming languages, libraries and frameworks may be evolving fast,
research software tends to be nurtured and developed by generations of researchers,
often without formalized code management practices (\gls{DIST}).
Experienced RSEs help modernising code, if necessary making it leaner and faster,
often via work-intensive refactoring, and possibly exploiting code transformation tools.
Notice the overlap with the \ac{HPC} RSE.

#### Web-development RSE
Web services have to offer users (which may include the public) a usable and accessible interface and yet be functional, interoperable in the provided data formats, whilst also offering security and data integrity.
The RSE balancing these contrasting needs communicates the trade-offs made with the involved stakeholders.

> As elsewhere, adhering to the values in @sec:values provides guidance.

Author: ```suggestion
```

# RSE specialisations {#sec:rse-specialisations}

> Now that we have defined base RSE skills,
> we list here typical specializations we encounter.
> Most RSEs roles embody a blend of these, and may stretch beyond.

Reviewer: ```suggestion
There is a large variety of RSE roles that embody a blend of the basic skills and competencies defined above.
We now list some of these specialisations.
```

Reviewer 2: Good suggestion!

## Specialisations within the core RSE competencies

#### Open science RSE

> Open science practices are not only being adopted increasingly, but also required by research funding agencies.

Reviewer: ```suggestion
Open science practices are increasingly required by research funding agencies.
```

The Open Science RSE may accompany international teams sharing large code bases, ensuring FAIRness of the code (\gls{SRU}, \gls{SP}, \ac{RDM}) in accordance to their research plans and cycles (\gls{RC}).

#### Project/community manager RSEs
Scaling up a research software project size is smoother if somebody
takes an overview role, caring about the overall perspective (\gls{PM})

> and involved people in their different roles (e.g. \gls{USERS}, and \gls{TEAM}).

Reviewer: _involved_ is the wrong tense, maybe

```suggestion
and delegating tasks to people in their different roles (e.g. \gls{USERS}, and \gls{TEAM}).
```

> Scaling up a research software project size is smoother if somebody
> takes an overview role, caring about the overall perspective (\gls{PM})
> and delegating tasks to people in their different roles (e.g. \gls{USERS}, and \gls{TEAM}).
> This role may or may not involve programming.

Reviewer: ```suggestion
Large research software projects require somebody to take
an overview role, caring about the overall perspective (\gls{PM})
and delegating tasks to project members (e.g. \gls{USERS}, and \gls{TEAM}).
This role may or may not involve programming.
```

Reviewer 2: Good suggestion!

#### Teaching RSEs

> Teaching (\gls{TEACH}) the next generation of scientists (RSEs or not) the core RSE skills and yet respecting the domain's culture requires having rudiments in didactics and pedagogy, and prepares the terrain for better research software to come.

Reviewer: ```suggestion
Teaching (\gls{TEACH}) the next generation of scientists (RSEs or not) the core RSE skills and yet respecting the domain's culture requires being versed in didactics and pedagogy. It prepares the scene for better research software to come.
```

> Teaching (\gls{TEACH}) the next generation of scientists (RSEs or not) the core RSE skills and yet respecting the domain's culture requires being versed in didactics and pedagogy. It prepares the scene for better research software to come.

Reviewer: ```suggestion
Teaching (\gls{TEACH}) the next generation of scientists (RSEs or not) the core RSE skills while respecting the domain's culture requires being versed in didactics and pedagogy. Educators play
a vital role in improving the quality of research software.
```

Reviewer 2: Good suggestion!

#### User interface/user experience designers for research software

> Good software is not only functional, but also has proper documentation (\gls{DOCBB}), is properly distributed (\gls{DIST}, \ac{FAIR}) and is ergonomic (\gls{MOD}).

Reviewer: ```suggestion
Good software is not only functional, but is also documented (\gls{DOCBB}) and distributed distributed (\gls{DIST}, \ac{FAIR}). It is also designed with users in mind (\gls{MOD}).
```

Author: there's a typo in the suggestion.

Reviewer: hm, I think I must have fixed the typo. At least I cannae spot it.

Reviewer 2: there's a double 'distributed'

Reviewer: aha, that's not something the spell checker can find and I am not plugged into an AI (yet?) - updated my suggestion

> Some RSEs devote themselves to these aspects.

Reviewer: I'd just drop this sentence.

Reviewer 2: Agree

Author: did just that.

> Some RSEs devote themselves to these aspects.

Reviewer: ```suggestion
```

## Specialisations outside the core RSE competencies

<!--
Research focused specialisations
-->

#### \${DOMAIN}-RSE
Some RSEs may be particularly specialized into the intricacies of a research domain,
(e.g. medical, humanities, physics).

#### Data-focused RSE
Certain domains have a high demand for data science skills.
This may be cleaning, sorting, analyzing and documenting data (\ac{RDM}, \ac{DMP}),

> perhaps anonymizing it (e.g. patient information in medicine),

Reviewer: ```suggestion
perhaps anonymising or pseudonymising them (e.g. patient information in medicine).
```

> and most often transferring securely and archiving properly \ac{FAIR} [@FAIR].

Reviewer: ```suggestion
This role also includes transferring data securely and archiving them properly \ac{FAIR} [@FAIR].
```

Author: shouldn't it be it?

Reviewer 2: I think data is a difficult one. 
We could maybe disembiguate by writing `data sets`

<!--
New areas of expertise
-->

#### Research infrastructure RSE
Specialized scientific software is increasingly accessible via services residing on the Internet.

> Configuring them requires knowledge also in administration, hardware, user management, access permissions (\glspl{SysOp}, \ac{IT}, \ac{ML}).

Reviewer: ```suggestion
Configuring these services also requires knowledge in administration, hardware, user management and access permissions (\glspl{SysOp}, \ac{IT}, \ac{ML}).
```

If evaluating access proposals to resources, they judge energy and efficiency
aspects, which next to scientific relevance is of increasing importance.
If working closer to users, they provide specialized knowledge for either manual or tool-assisted
\ac{HPC}-oriented refactoring, in collaborations focused on performance and porting.

#### ML-RSE
Machine learning (\ac{ML}) techniques stem from mathematical optimisation and statistics.
Software for \ac{ML} can be quite high level in its usage interface, and yet have \ac{HPC}-grade internals.
This motivates the need for an ML-RSE to not only know the given application domain (feature engineering) and the related software (be it for image data, linguistic data, etc) but also to chose the most adequate method combination (e.g. deep learning, reinforcement learning, neuro-symbolic learning, etc)
Since \ac{ML} techniques are inherently prone to biases leading to socially problematic consequences, this calls for additional caution, especially in training data.

#### Legacy RSEs
While programming languages, libraries and frameworks may be evolving fast,

> research software tend to be nurtured and passed over generations of researchers,

Reviewer: ```suggestion
research software tends to be nurtured and developed by generations of researchers,
```

> and often without formalized code management practices (\gls{DIST},...).

Reviewer: ```suggestion
often without formalized code management practices (\gls{DIST},...).
```

> often without formalized code management practices (\gls{DIST},...).

Reviewer: ```suggestion
often without formalized code management practices (\gls{DIST}).
```
drop ellipses

> Experienced RSEs help overcoming deprecated practices and features, if necessary making code leaner and faster,

Reviewer: ```suggestion
Experienced RSEs help modernising code, if necessary making it leaner and faster,
```

often via work-intensive refactoring, and possibly exploiting code transformation tools.
Notice the overlap with the \ac{HPC} RSE.

#### Web-development RSE

> Web services have to offer users (which may include the public) an usable and accessible interface and yet be functional, interoperable in the provided data formats, and yet offering security and data integrity.

Reviewer: ```suggestion
Web services have to offer users (which may include the public) a usable and accessible interface and yet be functional, interoperable in the provided data formats, whilst also offering security and data integrity.
```

The RSE balancing these contrasting needs communicates the trade-offs made with the involved stakeholders.

> As elsewhere, adhering to the values in @sec:values provides guidance.

Reviewer: ```suggestion
As elsewhere, they adhere to the values in @sec:values provides guidance.
```

Author: they adhere to X provides guidance, doesn't sound right either....

Reviewer: we could just drop that sentence

Reviewer 2: Yes, let us drop it

Author: did just that.

#### Legal-RSE
Creating, composing, and then publishing research software having different sources and licenses has legal constraints.

> With the advent of data protection regulatory frameworks (e.g. \ac{GDPR} in the \ac{EU} [@GDPR]),
> and for compliance with export control regulations (e.g. [@ExportControl] in Germany),

Reviewer: ```suggestion
With the advent of data protection regulatory frameworks,
and for compliance with export control regulations,
```
drop examples to reduce number of references
